### PR TITLE
Changes on travis build matrix

### PR DESCRIPTION
**Pull request by nerzhul**

### CODE REVIEW

File: util/travis/script.sh [hunk 1]
```diff
@@ -1,9 +1,16 @@
 #!/bin/bash -e
 
-if [[ $PLATFORM == "Unix" ]]; then
+if [[ $PLATFORM == "Unix" ]] || [[ $PLATFORM == "UnixCpp11" ]]; then
 	mkdir -p travisbuild
 	cd travisbuild || exit 1
 	CMAKE_FLAGS=''
+	if [[ $PLATFORM == "UnixCpp11" ]]; then
+		CMAKE_FLAGS+=" -DCMAKE_CXX_FLAGS=-std=c++11"
+		if [[ $CC == "gcc" ]]; then
```
Comment (nerzhul) [inline util/travis/script.sh]: yes i can, but if i add clang 3.8 ? :)

File: util/travis/before_install.sh [hunk 2]
```diff
@@ -8,6 +8,8 @@ if [[ $TRAVIS_OS_NAME == "linux" ]]; then
 		sudo apt-get update
 		sudo apt-get install llvm-3.1
 		sudo apt-get install clang
+	elif [[ $CC == "gcc" ]] && [[ $PLATFORM == "UnixCpp11" ]]; then
+		sudo apt-get install -y g++-6 gcc-6
 	fi
 	sudo apt-get update
 	sudo apt-get install p7zip-full
```
Comment (sfan5) [inline util/travis/before_install.sh]: Instead:

* Change this to `sudo apt-get install p7zip-full $COMPILER`
* Remove the distinction between `Unix` and `UnixCpp11`
* Add `COMPILER=g++` to the normal targets
* Make the script.sh check for `$COMPILER == "g++-6"`

Comment (nerzhul) [inline util/travis/before_install.sh]: Fixed

File: .travis.yml [hunk 3]
```diff
@@ -1,29 +1,31 @@
...
   email: false
 matrix:
   fast_finish: true
-  exclude:
-    - env: PLATFORM=Win32
+  include:
+    - env: PLATFORM=Win32 COMPILER=g++
+      compiler: gcc
+      os: linux
+    - env: PLATFORM=Win64 COMPILER=g++
```
Comment (sfan5) [inline .travis.yml]: windows should work without `COMPILER=g++`

File: util/travis/script.sh [hunk 4]
```diff
@@ -1,9 +1,15 @@
 #!/bin/bash -e
 
-if [[ $PLATFORM == "Unix" ]]; then
+if [[ $PLATFORM == "Unix" ]] || [[ $PLATFORM == "UnixCpp11" ]]; then
```
Comment (sfan5) [inline util/travis/script.sh]: remember to remove this addition

File: util/travis/script.sh [hunk 5]
```diff
@@ -1,9 +1,15 @@
...
 	mkdir -p travisbuild
 	cd travisbuild || exit 1
 	CMAKE_FLAGS=''
+	if [[ $COMPILER == "g++-6" ]]; then
+		if [[ $CC == "gcc" ]]; then
```
Comment (sfan5) [inline util/travis/script.sh]: will always be true

**Pull request merged**